File: Java/백준/Bronze/1598. 꼬리를 무는 숫자 나열/꼬리를 무는 숫자 나열.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String[] input = br.readLine().split(" ");
        int N = Integer.parseInt(input[0]);
        int M = Integer.parseInt(input[1]);

        int nRow = getRow(N);
        int nCol = getCol(N);
        int mRow = getRow(M);
        int mCol = getCol(M);

        int distance = Math.abs(nRow - mRow) + Math.abs(nCol - mCol);
        System.out.println(distance);
    }

    public static int getRow(int num) {
        return (num - 1) / 4 + 1;
    }

    public static int getCol(int num) {
        return (num - 1) % 4 + 1;
    }
}

File: C11/백준/Bronze/1598. 꼬리를 무는 숫자 나열/꼬리를 무는 숫자 나열.c
#include <stdio.h>
#include <stdlib.h>

int getRow(int num) {
    return (num - 1) / 4 + 1;
}

int getCol(int num) {
    return (num - 1) % 4 + 1;
}

int main() {
    int N, M;
    scanf("%d %d", &N, &M);

    int nRow = getRow(N);
    int nCol = getCol(N);
    int mRow = getRow(M);
    int mCol = getCol(M);

    int distance = abs(nRow - mRow) + abs(nCol - mCol);
    printf("%d\n", distance);

    return 0;
}